101-automation-configuration/azuredeploy.json

```diff
@@ -217,28 +248,36 @@
             "value": "[parameters('adminUsername')]"
           },
           "adminPassword": {
-            "value": "[parameters('adminPassword')]"
+            "reference": {
+              "keyVault": {
+                "id": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().Name, '/providers/Microsoft.KeyVault/vaults/', variables('keyVaultName'))]"
+              },
+              "secretName": "VMPassword"
+            }
           },
           "NicName": {
             "value": "[variables('NicName')]"
           },
-          "automationAccountName": {
-            "value": "[variables('automationAccountName')]"
+          "registrationUrl": {
+            "value": "[reference('provisionConfiguration', '2016-09-01').outputs.registrationUrl.value]"
```

Comment:
update apiVersion

Reply:
AutomationAccount's newest api is from 2015

Reply:
you're referencing Microsoft.Resources/deployments